lib/vehicle.rb

```diff
@@ -1,25 +1,54 @@
 require 'date'
 
 class Vehicle
-  attr_reader :vin,
-              :year,
-              :make,
-              :model,
-              :engine
 
-  def initialize(vehicle_details)
-    @vin = vehicle_details[:vin]
-    @year = vehicle_details[:year]
-    @make = vehicle_details[:make]
-    @model = vehicle_details[:model]
-    @engine = vehicle_details[:engine]
-  end
+    attr_reader :vin, 
+                :year, 
+                :make, 
+                :model,
+                :engine, 
+                :registration_date,
+                :plate_type,
+                :fees
 
-  def antique?
-    Date.today.year - @year > 25
-  end
+    def initialize(vehicle_details)
+        @vin = vehicle_details[:vin]
+        @year = vehicle_details[:year]
+        @make = vehicle_details[:make]
+        @model = vehicle_details[:model]
+        @engine = vehicle_details[:engine]
+        @registration_date
+        @plate_type
+        @fees = 0
+    end
 
-  def electric_vehicle?
-    @engine == :ev
-  end
-end
+    # Change Registration Date
+    def change_registration_date
+        @registration_date = Date.today
+    end
+
+    # Update Plate Type
+    def update_plate_type
+        if antique?
+            @plate_type = :antique
+            @fees += 25
+        elsif electric_vehicle?
+            @plate_type = :ev
+            @fees += 200
+        else
+            @plate_type = :regular
+            @fees += 100
+        end
```

Comment:
Hmmm... I'm not sure I like the fees being stored here on the Vehicle object. I think that information would be more dependent on the Facility, like different states/counties will charge different amounts, so I'd rather have a method in the Facility class that will determine the fees based on the Vehicles plate type. Good control flow logic though.